### ISSUE TITLE
Cannot start dataplaneapi without dataplaneapi.yaml

### ISSUE THREAD

**Issue by schegi** [haproxytech/dataplaneapi#175]
New configuration file seems to be mandatory to start the dataplane api, if it doesn't exist starting the api fails with time="2021-03-01T16:11:21Z" level=fatal msg="open dataplaneapi.yaml: no such file or directory".

**Comment by oktalz** [comment]
yes. this was actually a bug, since introduction of service discovery, we need to have file to save data you can post for it (consul api)

otherwise what you post on that endpoint will only work until you restart dataplane
this was resolved in https://github.com/haproxytech/dataplaneapi/commit/3f72efde8133a470057daf124c9b0df96e96552b (that will be backported to 2.2)
but on 2.3 having that file will be mandatory

**Comment by oktalz** [comment]
on 2.2 it can work without it (just create it on the fly if needed) but for 2.3 curent design requires that you create a file

structure of the file can be seen here https://github.com/haproxytech/dataplaneapi/blob/master/configuration/README.md
you can pick between yaml & hcl

documentation bout that is present but needs to be improved :)

**Comment by schegi** [comment]
Thx.

**Comment by schegi** [comment]
Build a new image this morning, now it works without the config file. But if i provide one starting the dataplane api always fails with 'time="2021-03-02T09:36:07Z" level=fatal msg="Error initiating users: no users configured in /etc/haproxy/haproxy.cfg, error: section missing"'. I can confirm that a user is provided in the haproxy.cfg

Btw, yesterday i tried to configure logging using the config file, i did not succeed. adding a section log: works so far but if i try to add, e.g. logto: file. After starting the dataplaneapi the config is rewritten and only features an empty log: {} section. I am not quite shure how i can find out the right config parameters.

**Comment by secretlyvogon** [comment]
I am facing the same issue as described.

DataPlane API version:
```HAProxy Data Plane API v2.2.0 75e0ae7.dev```

HaProxy version: 
```HA-Proxy version 2.0.13-2ubuntu0.1 ```

Could you also explain how/where should I mention the yaml config file in testing dataplane API. I am following steps from the documentation here: https://www.haproxy.com/documentation/hapee/latest/api/data-plane-api/installation/haproxy-community/

**Comment by schegi** [comment]
If you are using the referenced documentation it should be sufficient to have a dataplaneapi.yaml file right in the location you start the dataplaneapi from. Alternatively you can put it where you like and add -f path_to_file to your dataplaneapi call.

**Comment by oktalz** [comment]
yes, as @schegi said you can create (even an empty) dataplaneapi.yaml and call dataplane with additional params

```
dataplaneapi -f dataplaneapi.yaml .... rest of the arguments 
```
order is not important, but file needs to be there.

Note that for latest released version `2.2.0` you do not need to have this file, this will be requirement from 2.3+
